homeassistant/components/recollect_waste/__init__.py

```diff
@@ -64,9 +66,18 @@ async def async_get_pickup_events() -> List[PickupEvent]:
             hass.config_entries.async_forward_entry_setup(entry, component)
         )
 
+    hass.data[DOMAIN][DATA_LISTENER][entry.entry_id] = entry.add_update_listener(
+        async_reload_entry
+    )
+
     return True
 
 
+async def async_reload_entry(hass, config_entry):
+    """Handle an options update."""
+    await hass.config_entries.async_reload(config_entry.entry_id)
```

Comment:
IIRC these are weak references so they should be removed automatically

Reply:
Meaning I don't need to explicitly hold onto a cancel callback? Or something else I should go here?

Reply:
Thats how I remember it :)